### PR TITLE
Run ec2_instance tests in parallel and re-enable them

### LINKED ISSUE #58650: unstable ec2_instance integration tests
##### SUMMARY
Tracker for unstable ec2_instance integration tests. As of #56313 these tests run in CI. Here are a handful of the errors I'm periodically seeing.

https://app.shippable.com/github/ansible/ansible/runs/130541/103/tests

test/integration/targets/ec2_instance/tasks/checkmode_tests.yml:11
```
Traceback (most recent call last):
  File "/tmp/ansible_ec2_instance_payload_5vf1ee2g/__main__.py", line 1024, in discover_security_groups
    sub = ec2.describe_subnets(SubnetIds=[subnet_id])
  File "/usr/local/lib/python3.6/dist-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/botocore/client.py", line 661, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidSubnetID.NotFound) when calling the DescribeSubnets operation: The subnet ID 'subnet-018097bd633d16bc6' does not exist
```
test/integration/targets/ec2_instance/tasks/block_devices.yml:35
```
{
    "msg": "The conditional check 'in_test_vpc_instance.instances[0].block_device_mappings[0]' failed. The error was: error while evaluating conditional (in_test_vpc_instance.instances[0].block_device_mappings[0]): list object has no element 0"
}
```

https://app.shippable.com/github/ansible/ansible/runs/130541/102/tests

/test/integration/targets/ec2_instance/tasks/external_resource_attach.yml:60
```
{
"msg": "The conditional check 'in_test_vpc_instance.instances.0.key_name == \"shippable-130541-102_test_key\"' failed. The error was: error while evaluating conditional (in_test_vpc_instance.instances.0.key_name == \"shippable-130541-102_test_key\"): list object has no element 0"
}
```

https://app.shippable.com/github/ansible/ansible/runs/130511/102/tests

test/integration/targets/ec2_instance/tasks/cpu_options.yml:83
```
{
    "assertion": "'terminated' in ['running','pending']", 
    "changed": false, 
    "evaluated_to": false, 
    "msg": "Assertion failed"
}
```

https://app.shippable.com/github/ansible/ansible/runs/130542/103/tests

test/integration/targets/ec2_instance/tasks/external_resource_attach.yml:60
```
{
    "assertion": "(in_test_vpc_instance.instances.0.network_interfaces | length) == 1",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```

https://app.shippable.com/github/ansible/ansible/runs/130542/102/tests

test/integration/targets/ec2_instance/tasks/ebs_optimized.yml:31
```
{
    "msg": "The conditional check 'in_test_vpc_instance.instances[0].block_device_mappings[0]' failed. The error was: error while evaluating conditional (in_test_vpc_instance.instances[0].block_device_mappings[0]): list object has no element 0"
}
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
test/integration/targets/ec2_instance

##### ANSIBLE VERSION
```
ansible 2.9.0.dev0
```

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY
Try running the ec2_instance tests in parallel, to avoid aws-terminator coming to get us.

Fixes: #58650

##### ISSUE TYPE
- Bugfix Pull Request
(Tests bugfix)

##### COMPONENT NAME
lib/ansible/modules/cloud/amazon/ec2_instance.py

##### ADDITIONAL INFORMATION
~~Beware, tries to spin up 10 VPCs at a time, this is more that the default limit.~~

### CODE REVIEW

**Comment by tremble** [comment]
@s-hertel : initial attempt at running the tests in parallel
@mattclay : You asked about this yesterday as I was preparing to head home, so I didn't notice and respond...

**Comment by tremble** [comment]
@s-hertel I believe this is working (works consistently in my environment...)

File: test/integration/targets/ec2_instance/runme.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# Beware: most of our tests here are run in parallel.
+# To add new tests you'll need to add a new host to the inventory and a matching
+# '{{ inventory_hostname }}'.yml file in roles/ec2_instance/tasks/
+
+
+set -eux
+
+export ANSIBLE_ROLES_PATH=../
+
+ansible-playbook main.yaml -i inventory "$@"
```
Comment (jillr) [inline test/integration/targets/ec2_instance/runme.sh]: For consistency can you rename main.yaml to main.yml?

File: test/integration/targets/ec2_instance/inventory [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[tests]
+#dummy
```
Comment (s-hertel) [inline test/integration/targets/ec2_instance/inventory]: This PR looks great to me. There are a few things that are commented out that could just be removed such as this (and the corresponding file).

File: test/integration/targets/ec2_instance/roles/ec2_instance/defaults/main.yml [hunk 3]
```diff
@@ -1,12 +1,14 @@
 ---
 # defaults file for ec2_instance
-ec2_instance_name: '{{ resource_prefix }}-node'
 ec2_instance_owner: 'integration-run-{{ resource_prefix }}'
 ec2_instance_type: 't3.micro'
+ec2_instance_tag_TestId: '{{ resource_prefix }}-{{ inventory_hostname }}'
 ec2_ami_name: 'amzn2-ami-hvm-2.*-x86_64-gp2'
 
 vpc_name: '{{ resource_prefix }}-vpc'
+#vpc_name: '{{ resource_prefix }}-vpc-{{ inventory_hostname }}'
```
Comment (s-hertel) [inline test/integration/targets/ec2_instance/roles/ec2_instance/defaults/main.yml]: You could remove this and the commented out vpc_seed below.

**Review by s-hertel — APPROVED** [review]

**Review by jillr — COMMENTED** [review]
One nit aside from s-hertel's comments, otherwise this is also working for me locally.

**Review by s-hertel — APPROVED** [review]
shipit!

**Review by jillr — APPROVED** [review]

**Pull request merged**